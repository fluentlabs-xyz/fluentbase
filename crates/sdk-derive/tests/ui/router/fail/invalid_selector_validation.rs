#![allow(dead_code)]
extern crate alloc;
extern crate fluentbase_sdk;
use alloc::string::String;
use fluentbase_sdk::{
    basic_entrypoint,
    derive::{router, Contract},
    SharedAPI,
};

#[derive(Contract)]
struct App<SDK> {
    sdk: SDK,
}

pub trait RouterAPI {
    fn greeting(&self, message: String) -> String;
}

#[router(mode = "solidity")]
impl<SDK: SharedAPI> RouterAPI for App<SDK> {
    // This should fail because the selector 0xDEADBEEF doesn't match
    // the calculated selector for greeting(string)
    #[function_id("0xDEADBEEF", validate(true))]
    fn greeting(&self, message: String) -> String {
        message
    }
}

impl<SDK: SharedAPI> App<SDK> {
    pub fn deploy(&self) {}
}

basic_entrypoint!(App);
