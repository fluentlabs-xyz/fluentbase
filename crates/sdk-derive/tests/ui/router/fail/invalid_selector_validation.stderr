error: Function ID mismatch: Expected 0xf8194e48 for 'greeting(string)', but got 0xdeadbeef
  --> tests/ui/router/fail/invalid_selector_validation.rs:24:19
   |
24 |     #[function_id("0xDEADBEEF", validate(true))]
   |                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: You're seeing this error because you have validation enabled (validate(true))
   = help: To fix this, you can either:
   = help: 1. Use the expected function ID: #[function_id("greeting(string)")]
   = help: 2. Remove the validate parameter entirely: #[function_id("your_signature")]
   = help: 3. Or explicitly disable validation: #[function_id("your_signature", validate(false))]
